BlueprintUI/Sources/Layout/GridRow.swift

```diff
@@ -0,0 +1,191 @@
+            guard items.count > 0 else {
+                return .zero
+            }
+
+            // All frames share a height. Use the first one.
+            let height = _frames(in: constraint.width.maximum, for: items)[0].height
+
+            return CGSize(width: constraint.width.maximum, height: height)
+        }
+
+        func layout(size: CGSize, items: [(traits: Width, content: Measurable)]) -> [LayoutAttributes] {
+            guard items.count > 0 else {
+                return []
+            }
+
+            return _frames(in: size.width, for: items)
+                .map(LayoutAttributes.init)
+        }
+
+        private func _frames(
+            in width: CGFloat,
+            for items: [(traits: Width, content: Measurable)]
+        ) -> [CGRect] {
+            var sizes: [CGSize] = (0..<items.count).map { _ in .zero }
+
+            // Group children by their sizing. Maintain child order by also storing index.
+            var absolutelySized: [(index: Int, width: CGFloat, content: Measurable)] = []
+            var proportionallySized: [(index: Int, proportion: CGFloat, content: Measurable)] = []
+            items.enumerated().forEach { (index, item) in
+                switch item.traits {
+                case .absolute(let width):
+                    absolutelySized.append((index, width, item.content))
+                case .proportional(let proportion):
+                    proportionallySized.append((index, proportion, item.content))
+                }
+            }
+
+            // Measure the height of absolutely-sized children and record the total width they require.
+            var absoluteItemWidth: CGFloat = 0
+            absolutelySized.forEach { (index, width, content) in
+                absoluteItemWidth += width
+                let height = content.measure(in: SizeConstraint(width: width)).height
+                sizes[index] = CGSize(width: width, height: height)
+            }
+
+            // Measure proportionally-sized children, allotting them a portion of the available space.
+            // Determine the available width.
+            let availableWidth = width - (spacing * CGFloat(items.count - 1)) - absoluteItemWidth
+            // Determine the number of portions and the width of each portion.
+            let portionCount = proportionallySized.map { $0.proportion }.reduce(0, +)
+            let portion = availableWidth / portionCount
+            // Assign a proportional width to each proportionally-sized child.
+            proportionallySized.forEach { (index, proportion, content) in
+                let width = portion * proportion
+                let height = content.measure(in: SizeConstraint(width: width)).height
+                sizes[index] = CGSize(width: width, height: height)
+            }
+
+            // Use the tallest child to set the height of all children.
```

Comment:
Would it make sense to allow a way to align the content top/bottom/centered here?

Reply:
I'm open to it. As is, one would have to wrap children in `Alignment` elements to achieve this effect, which isn't particularly nice.

Reply:
I added this feature in the following commit. The additional layout complexity is actually not too bad:

```swift
// Begin computing frames by reading the vertical alignment. Vertical alignment determines the y-origin
// and, in the `.fill` case, the height of the children.
// The tallest child will be used to size the row.
let rowHeight = sizes.lazy.map { $0.height }.max() ?? 0
let yVector: (CGSize) -> (origin: CGFloat, height: CGFloat)
switch verticalAlignment {
case .fill:
    // Use the tallest child to set the height of all children.
    yVector = { _ in (origin: 0, height: rowHeight) }
case .top:
    yVector = { size in (origin: 0, height: size.height) }
case .center:
    yVector = { size in (origin: (rowHeight - size.height) / 2.0, height: size.height) }
case .bottom:
    yVector = { size in (origin: rowHeight - size.height, height: size.height) }
}
```

Basically, we need to update `yOrigin` and `height` based on alignment. Additionally, we now need to read all frames to determine row height.

I used `lazy` because I hoped it would lead to smaller allocations, but I'm happy to cut it out.